pkgs/btcpayserver/update

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+
+set -e
+
+success=0
+worktrees=()
+here="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+cleanup() {
+  printf 'cleaning up...\n'
+  rm -rf "${worktrees[@]}"
+
+  if [ $success -eq 1 ]; then
+    printf "now type:\n\n    nix-build -E 'with import <nixpkgs> {}; callPackage ./btcpayserver.nix {}'\n\nor\n\n"
+    printf "    nix-build -E 'with import <nixpkgs> {}; callPackage ./nbxplorer.nix {}'\n\nto see if it worked\n"
+  fi
+}
+trap cleanup EXIT
+
+usage() {
+  printf 'usage: ./update <BTCPayServer version> <NBXplorer version>\n\n'
+  printf '  eg: ./update v1.0.4.4 v2.1.26\n'
+  exit 1
+}
+
+if [ -z "$1" ] || [ -z "$2" ]; then
+  usage
+fi
+
+printf 'getting utils...\n'
+dotnet2nixpath="$(nix-build --no-out-link 'https://github.com/jb55/dotnet2nix/archive/latest.tar.gz')/bin"
+dotnetpath=$(nix-build --no-out-link '<nixpkgs>' -A dotnet-sdk_3)/bin
+nix_prefetch_github=$(nix-build --no-out-link '<nixpkgs>' -A nix-prefetch-github)/bin
+export PATH=$nix_prefetch_github:$dotnetpath:$dotnet2nixpath:$PATH
+
+process_repo() {
+  local owner
+  local repo
+  local rev
+  local worktree
+  local checkout
+
+  owner=$1
+  repo=$2
+  rev=$3
+
+  shift
+  shift
+  shift
+
+  checkout=${CHECKOUT_DIR:-/tmp}/${repo}-update
+  worktree="$(mktemp -d)"
+  worktrees+=("$worktree")
+
+  if [ ! -d "$checkout" ]
+  then
+    printf "cloning %s/%s...\n" "$owner" "$repo"
+    git clone --bare "https://github.com/$owner/$repo" "$checkout"
+  else
+    printf "found existing %s bare checkout @ %s\n" "$repo" "$checkout"
+  fi
+
+  printf 'creating temporary worktree for %s at %s\n' "$rev" "$worktree"
+
+  cd "$checkout"
+  git fetch origin "$rev"
+  git worktree add "$worktree" FETCH_HEAD
+
+  cd "$worktree"
+
+  dotnet restore -r linux-x64
+  dotnet2nix "$@"
+
+  cp nuget-packages.json "$here"/"$repo"-packages.json
+
+  ver=$rev
+  if [ "${rev::1}" == "v" ]; then
+    ver=${rev:1}
+  fi
+
+  sha256=$(nix-prefetch-github "$owner" "$repo" --rev "$rev" | jq -r .sha256)
```

Comment:
What do you think about using fetchurl and calculating the sha256 with `git archive` from Nicolas Dorier's signed git commits (0x6618763EF09186FE)? Similar to `pkgs/electrs/default.nix` and `pkgs/electrs/get-sha256.sh`.

Reply:
oh yeah this would be easy to do, I think I had this for testing git revisions. We should just be doing nix-prefetch-url with signed tarballs instead